teps/0052-tekton-results-automated-run-resource-cleanup.md

```diff
@@ -73,26 +109,53 @@ Relevant Issues:
 
 ### Goals
 
+- Separate concerns for users who need current runtime data vs users who need
+  historical result data.
 - Define a mechanism for cleaning up completed Runs to minimize resources the
-  Pipeline controller is responsible for.
-- Preserve Run metadata so that users can continue to query it beyond the
-  lifetime of the Pipeline controller.
+  Pipeline data plane stores.
+- Preserve Run objects (e.g. the entire CRD) so that users can continue to query
+  it beyond the lifetime of the Pipeline data plane.
 
 ### Non-Goals
 
-- Providing a programmatic mechanism for end users to dynamically control deletion behavior.
+- Providing a programmatic mechanism for end users to dynamically control
+  deletion behavior.
+- Preservation of Pod logs.
+- Defining retention policies for Result API data.
 
 ### Use Cases (optional)
 
-This is a commonly requested feature due to the strain Run history this puts on
-etcd over time. See [Relevant Issues under Motivation](#motivation).
+When interacting with Run results, we see 2 primary users -
+
+1. Runtime Operators - Users that need to know what's currently running in a
+   cluster. They require the most up-to-date data, but don't require knowledge
+   of previous events when complete. Example users:
+
+   - Tekton Pipeline Controller
+   - Tekton Results Watcher
+   - Third Party Controllers responding to Task/PipelineRun lifecycle events.
+
+2. Result "Historians" (struggling with naming here) - Users who want to make
+   queries about historical Run results. These users can tolerate short delays
+   in data population, but want to query over all previous runs. Example
+   queries:
+
+   - When did this daily Pipeline start failing?
+   - Show me the execution duration of all PipelineRuns that match a certain
+     condition.
+   - Show me all Runs related to a particular Git repo.
+
+   Users that might fall under this category:
+
+   - Build captains
+   - Auditors
 
 ## Requirements
 
 - Run metadata should be available even after deletion from the Pipeline
   controller data plane.
-- Runs should be deleted as soon as it can be done safely (i.e. we have
-  confirmed the data is stored durably elsewhere).
+- Runs should not be deleted until we have confirmed the data is stored durably
```

Comment:
I wonder if this kind of signal ("Run XYZ Persisted!") would be useful to many different kinds of systems, not just deletion.  Are there plans in the works already for Results to emit events like this for consumption by other services?

Reply:
No solid plans at the moment, but I think that's a discussion worth having w.r.t. [TEP-0032: Tekton Notifications](https://github.com/tektoncd/community/blob/main/teps/0032-tekton-notifications.md).

Reply:
@wlynch what would you think about including a requirement to emit a cloudevents during the results lifecycle? (or maybe even as part of the results TEP?) i think @sbwsg makes a great point

Reply:
I do want to explore this further, but I think this probably deserves its own TEP (or we co-opt the existing Tekton Notifications TEP)

/cc @afrittoli